### PR TITLE
Dfs alg

### PR DESCRIPTION
**Pull request by an78drew**
code review is required.
the code contains 2 features:
- dfs algorithm to
- come back to the nearest fork containing unvisited tiles

### CODE REVIEW

File: temple/src/main/java/game/GameState.java [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 import gui.GUI;
 import student.Explorer;
+import student.LogHelper;
```
Comment (djvcom) [inline temple/src/main/java/game/GameState.java:5]: I can see the class you've made for this, but are we actually using it in the code? I can't see that we are. I think we're probably ok logging in this class for debugging purposes as well, but I'm pretty sure any code changes in here will be diregarded when the coursework is graded. Perhaps that's not an issue though?

Comment (an78drew) [reply temple/src/main/java/game/GameState.java:5]: Hi, the helper is used for debugging for sure, we can remove it. Shall I do it for the current request?

Comment (djv3) [reply temple/src/main/java/game/GameState.java:5]: Maybe let's take it out for now (happy for it to stay on another separate open branch so we don't lose the code however). Otherwise this looks good to me, it can be a little hard to follow nested if/else logic, but I have no suggestions really on how it could be refactored, nice work!

Comment (an78drew) [reply temple/src/main/java/game/GameState.java:5]: I've added a task on trello to remove loghelper before release. In case we finally go with dfs-alg for escape phase, I'll refactor it

**Review by djv3 — APPROVED** [review]
Looks good to me Andrew, thanks for taking the time to put it together!
